### ISSUE TITLE
False positive for null pointer inspection when using nullsafe operator. 

### ISSUE THREAD

**Issue by michaeldnelson** [kalessil/phpinspectionsea#1768]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended)  4.0.6.4 |
| Language level | 8.0                  |

Current behaviour

These two functions are equivalent.  $t is perceived as nullable in testEA but is recognized correctly as not nullable in EA2.

--
<img width="562" alt="Screen Shot 2021-09-20 at 12 30 36 AM" src="https://user-images.githubusercontent.com/829489/133957280-8cddeddb-9643-4af0-8e75-d3d89d794a53.png">

Expected behaviour

$t should be recognized as non nullable in testEA

**Comment by EtienneBruines** [comment]
To add a potentially related use-case / variation:

```php
public function testEA(): void 
{
    $t = $this->nullOrT();
    if ($t) {}
    $t->isValid(""); // This line is not marked as potentially null
}
```

**Comment by ea-inspections-team** [comment]
Thanks for reporting @michaeldnelson. We looked into possible solutions, but we are limited with how IDE types resolving works here and can not provide a fix.

**Comment by ea-inspections-team** [comment]
@EtienneBruines : your use-case seems to be handled already (we were able to handle it properly) in older the versions of analyzer.
